src/lib/blocks.js

```diff
@@ -110,6 +110,75 @@ export default function (vm) {
         return [[myself, '_myself_']].concat(spriteMenu());
     };
 
+    const variablePropertyMenu = function (thisValue) {
+        // Fill the output array with sprite properties by default. This will be changed to stage properties if the
+        // stage is selected.
+        const xPosition = ScratchBlocks.ScratchMsgs.translate('SENSING_OF_XPOSITION', 'x position');
+        const yPosition = ScratchBlocks.ScratchMsgs.translate('SENSING_OF_YPOSITION', 'y position');
+        const direction = ScratchBlocks.ScratchMsgs.translate('SENSING_OF_DIRECTION', 'direction');
+        const costumeNumber = ScratchBlocks.ScratchMsgs.translate('SENSING_OF_COSTUMENUMBER', 'costume #');
+        const costumeName = ScratchBlocks.ScratchMsgs.translate('SENSING_OF_COSTUMENAME', 'costume name');
+        const size = ScratchBlocks.ScratchMsgs.translate('SENSING_OF_SIZE', 'size');
+        const volume = ScratchBlocks.ScratchMsgs.translate('SENSING_OF_VOLUME', 'volume');
+        let output = [
+            [xPosition, 'x position'],
+            [yPosition, 'y position'],
+            [direction, 'direction'],
+            [costumeNumber, 'costume #'],
+            [costumeName, 'costume name'],
+            [size, 'size'],
+            [volume, 'volume']
+        ];
+
+        const sourceBlock = thisValue.sourceBlock_; // This is the <shadow>.
+        const ofBlock = sourceBlock && sourceBlock.parentBlock_; // This is the "of" block.
+        if (ofBlock) {
+            let block;
+            let blocks;
+            if (vm.editingTarget) {
+                blocks = vm.editingTarget.blocks;
+                block = blocks.getBlock(ofBlock.id);
+            }
+            if (!block) {
+                // The block may be in the flyout.
+                blocks = vm.runtime.flyoutBlocks;
+                block = blocks.getBlock(ofBlock.id);
+            }
+            if (block) {
+                // Get the name of the sprite which is selected. This is based on the OBJECT dropdown's selected value,
+                // which may be different from the value returned by a block placed within that input, if present.
+                // TODO: Variables should probably not be fetched at all if the OBJECT dropdown contains a block.
```

Comment:
Making note of this. Actually the (whole) comment here is wrong - if you click the property dropdown while there is a block in the object dropdown, the editor stops working properly (the "of" block gets stuck to your cursor).

Reply:
Resolved - updated the code to not show anything if the slot is occupied by a block. (We find out if it is by checking if the block in the slot is a shadow block: if it is, the dropdown is empty; if it isn't, it's filled with a reporter.)